command-line/StatesAndEvents.py

```diff
@@ -0,0 +1,121 @@
+from enum import Enum
+
+from typing import List, Union
+
+from dataclasses import dataclass
+
+
+class VCUEvents (Enum):
+    """
+    The VCU Events, refer to VCU\Phantom\data_structures\vcu_common.h
+    """
+    EVENT_APPS1_RANGE_FAULT = 1
+    EVENT_APPS2_RANGE_FAULT = 2
+    EVENT_BSE_RANGE_FAULT = 3
+    EVENT_FP_DIFF_FAULT = 4
+    EVENT_RESET_CAR = 5
+    EVENT_READY_TO_DRIVE = 6
+    EVENT_TRACTIVE_ON = 7
+    EVENT_TRACTIVE_OFF = 8 
+    EVENT_BRAKE_PLAUSIBILITY_CLEARED = 9
+    EVENT_BRAKE_PLAUSIBILITY_FAULT = 10
+    EVENT_UNRESPONSIVE_APPS = 11
+class VCUStates(Enum):
+    """
+    The VCU states, refer to VCU\Phantom\data_structures\vcu_common.h
+    """
+    TRACTIVE_OFF = 0
+    TRACTIVE_ON = 1 
+    RUNNING = 2
+    MINOR_FAULT = 3
+    SEVERE_FAULT = 4
+
+@dataclass
+class EventData:
+    EVENT: VCUEvents
+    TIME: int
+
+@dataclass
+class StateData:
+    STATE: VCUStates
+    TIME: Union[int, None]
+
+class ResponseVCU:
+    """
+    The data structure in which the VCU returns values via UART communication
+    after sending a sequence of values
+
+    Allows for the same events to be stored if triggered at different times
+    It is necessary for the this class to specify the VCU state
+    
+    Stores an array of events and their time of trigger
+    Stores the state of the VCU and the time of change (if changed)
+    """
+
+    def __init__(self):
+        self._events: List[EventData] = []
+        self._state : StateData = None
+    
+    def add_event(self, event_name: VCUEvents, event_time: int):
+        """
+        Add an event to the list of events
+
+        :param event_name: a VCU event
+        :param event_time: the relative time of the event
+        """
+
+        #check that the event exists in the available VCU Events enum
+        if not isinstance(event_name, VCUEvents):
```

Comment:
Your implentation certainly is cleaner, but I like the explicit Raise that associates with an Event Error

I think it makes it reading the data structure more clear that it must be an VCU Event instead of getting the error implictly due to improper Enum cast